src/System.Threading/tests/MonitorTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using Xunit;
+
+namespace System.Threading.Tests
+{
+    static class MonitorTests
+    {
+        // Attempts a single recursive acqusition/release cycle of a newly-created lock.
+        [Fact]
+        public static void BasicRecursion(ref string message)
+        {
+            var obj = new object();
+            Assert.True(Monitor.TryEnter(obj));
+            Assert.True(Monitor.TryEnter(obj));
+            Monitor.Exit(obj);
+            Assert.True(Monitor.IsEntered(obj));
+            Monitor.Enter(obj);
+            Assert.True(Monitor.IsEntered(obj));
```

Comment:
It'd be good to add:
```C#
Monitor.Exit(obj);
Assert.True(Monitor.IsEntered(obj));
Monitor.Exit(obj);
Assert.False(Monitor.IsEntered(obj));
```
Otherwise we're not testing that the lock goes back to not being entered once the recursion unwinds completely.